packages/test-utils/src/storage-browser/mock-handlers.ts

```diff
@@ -137,17 +139,19 @@ export class MockHandlers {
       });
     }
 
-    if (
-      preventOverwrite &&
-      this.#locationItems[prefix].some((item) => item.key === key)
-    ) {
-      return {
-        ...UNDEFINED_CALLBACKS,
-        result: Promise.resolve({
-          error: new Error('cannot overwrite existing file'),
-          status: 'OVERWRITE_PREVENTED',
-        }),
-      };
+    if (this.#locationItems[prefix].some((item) => item.key === key)) {
+      if (preventOverwrite) {
+        return {
+          ...UNDEFINED_CALLBACKS,
+          result: Promise.resolve({
+            error: new Error('cannot overwrite existing file'),
+            status: 'OVERWRITE_PREVENTED',
+          }),
+        };
+      } else {
+        // remove prev item as new version will be added, 'overwriting' prev item
+        this.#locationItems[prefix].filter((item) => item.key !== key);
```

Comment:
AFAICT this line isn't accomplishing - `.filter` returns a shallow copy of the array, and does not modify the existing array

Reply:
Seems like it could be removed entirely?